### PR TITLE
Ws client fixes

### PR DESCRIPTION
**Pull request by ivmarkov**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -51,28 +51,48 @@ jobs:
         env:
           ESP_IDF_VERSION: ${{ matrix.idf-version }}
           ESP_IDF_SDKCONFIG_DEFAULTS: "${{ github.workspace }}/.github/configs/sdkconfig.defaults"
+<<<<<<< HEAD
           RUSTFLAGS: "${{ matrix.idf-version == 'v5.1.2' && '--cfg espidf_time64' || ''}}"
+=======
+          ESP_IDF_COMPONENT_MANAGER: "${{ matrix.idf-version == 'v5.1.1' && 'y' || 'n'}}"
+          RUSTFLAGS: "${{ matrix.idf-version == 'v5.1.1' && '--cfg espidf_time64' || ''}}"
```
Comment (Vollbrecht) [inline .github/workflows/ci.yml:58]: version 5.1.1 still in use here

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -51,28 +51,48 @@ jobs:
...
         run: cargo clippy --features nightly,experimental --no-deps --target ${{ matrix.target }} -Zbuild-std=std,panic_abort -Zbuild-std-features=panic_immediate_abort -- -Dwarnings
 
       - name: Build | Compile
         env:
           ESP_IDF_VERSION: ${{ matrix.idf-version }}
           ESP_IDF_SDKCONFIG_DEFAULTS: "${{ github.workspace }}/.github/configs/sdkconfig.defaults"
+<<<<<<< HEAD
           RUSTFLAGS: "${{ matrix.idf-version == 'v5.1.2' && '--cfg espidf_time64' || ''}}"
+=======
+          ESP_IDF_COMPONENT_MANAGER: "${{ matrix.idf-version == 'v5.1.1' && 'y' || 'n'}}"
```
Comment (Vollbrecht) [inline .github/workflows/ci.yml:69]: dito

File: components_esp32s3.lock [hunk 3]
```diff
@@ -0,0 +1,15 @@
+dependencies:
```
Comment (Vollbrecht) [inline components_esp32s3.lock:1]: we should add the .lock files produced by the component manager .gitignore?

**Review by Vollbrecht — CHANGES_REQUESTED** [review]
Added comments

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ivmarkov) [reply .github/workflows/ci.yml:58]: The code in the CI yaml will all be reverted. This example is prepared to only compile for ESP IDF 4. There are comments in the example if the user wants to build it for ESP IDF 5. But we cannot alter the `Cargo.toml` file of `esp-idf-svc` in a way where it becomes compatible with ESP IDF 5 only (by including the websocket component). The user will have to do this in their own `Cargo.toml.

File: components_esp32s3.lock (hunk 3 above)
Comment (ivmarkov) [reply components_esp32s3.lock]: There will be no lock files produced by `esp-idf-svc`, as per above.

**Comment by ivmarkov** [comment]
> Added comments

Thanks!

**Pull request merged**